src/test_helper.rs

```diff
@@ -0,0 +1,213 @@
+        &BoxConstraints::new(
+            kurbo::Size::ZERO,
+            kurbo::Size {
+                width: buffer_size.width.into(),
+                height: buffer_size.height.into(),
+            },
+        ),
+    );
+
+    let mut paint_cx = PaintCx {
+        cx_state: &mut cx_state,
+        widget_state: &mut widget_state,
+        terminal: &mut terminal,
+        override_style: Style::default(),
+    };
+
+    sut.paint(&mut paint_cx);
+    terminal.flush().unwrap();
+
+    terminal
+}
+
+/// This widget provides access to the terminal output of its children
+///
+/// After its children were painted it calls the flush() and clones the
+/// terminal's buffer and sends it using the passed [Sender<Buffer>].
+///
+/// This is handy for snapshot tests.
+pub struct DebugView<V, T, A> {
+    content: V,
+    debug_chan_tx: mpsc::Sender<Buffer>,
+    phantom: PhantomData<fn() -> (T, A)>,
+}
+
+pub fn debug_view<V, T, A>(content: V, debug_chan_tx: mpsc::Sender<Buffer>) -> DebugView<V, T, A> {
+    DebugView {
+        content,
+        debug_chan_tx,
+        phantom: PhantomData,
+    }
+}
+
+impl<T, A, V> ViewMarker for DebugView<V, T, A> {}
+
+impl<T, A, V> View<T, A> for DebugView<V, T, A>
+where
+    V: View<T, A>,
+    V::Element: 'static,
+{
+    type State = (V::State, xilem_core::Id);
+
+    type Element = DebugWidget;
+
+    fn build(&self, cx: &mut Cx) -> (xilem_core::Id, Self::State, Self::Element) {
+        let (id, (state, element)) = cx.with_new_id(|cx| {
+            let (child_id, state, element) = self.content.build(cx);
+            (
+                (state, child_id),
+                DebugWidget::new(element, self.debug_chan_tx.clone()),
+            )
+        });
+        (id, state, element)
+    }
+
+    fn rebuild(
+        &self,
+        cx: &mut Cx,
+        prev: &Self,
+        id: &mut xilem_core::Id,
+        (state, child_id): &mut Self::State,
+        element: &mut Self::Element,
+    ) -> crate::widget::ChangeFlags {
+        let changeflags = ChangeFlags::empty();
+
+        let element = element
+            .content
+            .downcast_mut()
+            .expect("The DebugView content widget changed its type, this should never happen!");
+
+        changeflags
+            | cx.with_id(*id, |cx| {
+                self.content
+                    .rebuild(cx, &prev.content, child_id, state, element)
+            })
+    }
+
+    fn message(
+        &self,
+        id_path: &[xilem_core::Id],
+        (state, child_id): &mut Self::State,
+        message: Box<dyn std::any::Any>,
+        app_state: &mut T,
+    ) -> MessageResult<A> {
+        if let Some((first, rest_path)) = id_path.split_first() {
+            if first == child_id {
+                self.content.message(rest_path, state, message, app_state)
+            } else {
+                xilem_core::MessageResult::Stale(message)
+            }
+        } else {
+            xilem_core::MessageResult::Nop
+        }
```

Comment:
This can just be 
```suggestion
        self.content.message(id_path, state, message, app_state)
```
then